source/agora/common/Transaction.d

```diff
@@ -365,68 +357,7 @@ unittest
         outputs : [Output(-400_000, key_pairs[1].address)]  // oops
     };
 
-    assert(!tx_2.verifyData(findOutput));
-}
-
-/// This creates a new transaction and signs it as a publickey
-/// of the previous transaction to create and validate the input.
-unittest
-{
-    import std.format;
-
-    Transaction[Hash] storage;
-
-    immutable(KeyPair)[] key_pairs;
-    key_pairs ~= KeyPair.random();
-    key_pairs ~= KeyPair.random();
-    key_pairs ~= KeyPair.random();
-
-    // delegate for finding `Output`
-    auto findOutput = (Hash hash, size_t index)
-    {
-        if (auto tx = hash in storage)
-            if (index < tx.outputs.length)
-                return &tx.outputs[index];
-
-            return null;
-    };
-
-    // Create the first transaction.
-    Transaction genesisTx = newCoinbaseTX(key_pairs[0].address, 100_000);
-    Hash genesisHash = hashFull(genesisTx);
-    storage[genesisHash] = genesisTx;
-    genesisTx.inputs[0].signature = key_pairs[0].secret.sign(hashFull(genesisTx)[]);
-
-    // Create the second transaction.
-    Transaction tx1 = Transaction(
-        [
-            Input(genesisHash, 0)
-        ],
-        [
-            Output(1_000, key_pairs[1].address)
-        ]
-    );
-
-    // Signs the previous hash value.
-    Hash tx1Hash = hashFull(tx1);
-    tx1.inputs[0].signature = key_pairs[0].secret.sign(tx1Hash[]);
-    storage[tx1Hash] = tx1;
-
-    assert(tx1.verifySignature(findOutput), format("Transaction signature is not validated %s", tx1));
-
-    Transaction tx2 = Transaction(
-        [
-            Input(tx1Hash, 0)
-        ],
-        [
-            Output(1_000, key_pairs[1].address)
-        ]
-    );
+    tx_2.inputs[0].signature = key_pairs[0].secret.sign(hashFull(tx_2)[]);
```

Comment:
It's weird to merge the signature verification check into a PR which documentation is:
> /// negative output amounts disallowed

Reply:
into a unittest you mean. Yeah I'll fix this up.